lms/djangoapps/discussion_api/api.py

```diff
@@ -138,6 +161,26 @@ def get_thread_list(request, course_key, page, page_size):
     # behavior and return a 404 in that case
     if result_page != page:
         raise Http404
+    staff_user_ids = {
+        user.id
+        for role in Role.objects.filter(
+            name__in=[FORUM_ROLE_ADMINISTRATOR, FORUM_ROLE_MODERATOR],
+            course_id=course_key
+        )
+        for user in role.users.all()
+    }
```

Comment:
i know it's early to be optimizing but this (and the following two) lists of ids are not user-specific and change infrequently, so i'd suggest at least adding a TODO about moving them to external, cacheable utils.